lib/improver/tests/cube_combiner/test_CubeCombiner.py

```diff
@@ -0,0 +1,266 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the windgust_diagnostic.WindGustDiagnostic plugin."""
+import unittest
+
+import numpy as np
+
+import iris
+from iris.tests import IrisTest
+from iris.cube import Cube
+from iris.coords import DimCoord
+from cf_units import Unit
+
+from improver.cube_combiner import CubeCombiner
+
+
+def create_cube_with_threshold(data=None,
```

Comment:
Yes it is very similar. Do we gain anything by using the existing function apart from another dependency on the helper functions?

Reply:
The main advantage would be to avoid duplicating these cube creation functions. Ideally, these sort of cube creation functions would be more centralised, rather than being in the `ensemble_calibration` or in neighbourhood processing, etc, so that we can be more confident that we're using cubes that have been created similarly. This functionality is noted in #50. For this PR, I don't mind whether this is changed, but I'm hopeful that one day we will invest the time to centralise these utilities, and making as much of common functions as is reasonable would be helpful for that process.